src/vorta/views/main_window.py

```diff
@@ -106,10 +108,46 @@ def set_progress(self, text=''):
         self.progressText.setText(text)
         self.progressText.repaint()
 
-    def set_log(self, text=''):
+    def set_log(self, text='', ret={}):
         self.logText.setText(text)
         self.logText.repaint()
 
+        msgid = ret.get('msgid')
+        repo_url = ret.get('repo_url')
+        cmd = ret.get('cmd')
+        if cmd is not None and cmd != 'init':
+            # Prevent getting 'None' profile on None case
+            profile = BackupProfileModel.get(name=ret['profile_name']) if ret.get('profile_name') is not None else None
+            if msgid == 'LockTimeout':
```

Comment:
Getting a log with `LockTimeout` or `LockFailed` will be very very rare. Still, you do a lot of work for every log message that will never be used:

```
        msgid = ret.get('msgid')
         repo_url = ret.get('repo_url')
         cmd = ret.get('cmd')
         ....
         profile = BackupProfileModel.get(name=ret['profile_name'])
```

So I recommend *only* looking at `msgid` first and doing every other operations *after* that.